Maple2.Server.Game/Model/Field/Actor/ActorStateComponent/AnimationState.cs

```diff
@@ -87,31 +92,61 @@ public struct TickPair {
 
         bool found = RigMetadata.Sequences.TryGetValue(name, out AnimationSequence? sequence);
 
-        ResetSequence();
+        if (reportingKeyframeEvent) {
+            queuedSequence = sequence;
+            queuedSequenceSpeed = speed;
+            queuedSequenceType = type;
+
+            return found;
+        }
 
         if (!found) {
-            DebugPrint($"Attempt to play nonexistent sequence '{name}' at x{speed} speed, previous: '{PlayingSequence?.Name ?? "none"}' x{sequenceSpeed}");
+            DebugPrint($"Attempt to play nonexistent sequence '{name}' at x{speed} speed, previous: '{PlayingSequence?.Name ?? "none"}' x{SequenceSpeed}");
+
+            if (reportingKeyframeEvent) {
+                queuedResetSequence = true;
+            } else {
+                ResetSequence();
+            }
 
             return false;
         }
 
-        DebugPrint($"Playing sequence '{sequence!.Name}' at x{speed} speed, previous: '{PlayingSequence?.Name ?? "none"}' x{sequenceSpeed}");
+        PlaySequence(sequence!, speed, type);
+
+        return true;
+    }
+
+    private void PlaySequence(AnimationSequence sequence, float speed, AnimationType type) {
+        if (PlayingSequence != sequence && actor is FieldNpc npc) {
+            npc.SendControl = true;
+        }
+
+        ResetSequence();
+
+        DebugPrint($"Playing sequence '{sequence!.Name}' at x{speed} speed, previous: '{PlayingSequence?.Name ?? "none"}' x{SequenceSpeed}");
 
         PlayingSequence = sequence;
-        sequenceSpeed = speed;
+        SequenceSpeed = speed;
         sequenceType = type;
 
         lastTick = actor.Field.FieldTick;
-
-        return true;
     }
 
     public void CancelSequence() {
         if (PlayingSequence is not null) {
-            DebugPrint($"Canceled playing sequence: '{PlayingSequence.Name}' x{sequenceSpeed}");
+            DebugPrint($"Canceled playing sequence: '{PlayingSequence.Name}' x{SequenceSpeed}");
         }
 
-        ResetSequence();
+        if (reportingKeyframeEvent) {
+            queuedResetSequence = true;
+        } else {
+            if (PlayingSequence != null && actor is FieldNpc npc) {
+                npc.SendControl = true;
+            }
+
+            ResetSequence();
+        }
```

Comment:
formatting might be off

```suggestion
        if (reportingKeyframeEvent) {
            queuedResetSequence = true;
            return;
        }
        if (PlayingSequence != null && actor is FieldNpc npc) {
            npc.SendControl = true;
        }

        ResetSequence();
```